### PR TITLE
implement a feature editor

### PR DESCRIPTION
**Pull request by bartvde**
This component will allow editing of feature attributes. If configured with a WFS protocol, it will allow saving the changes back to the server.

Ideally the commit action would be the responsibility of the FeatureStore, but this was a bridge too far for the purpose of this PR. We should definitely consider changing this in the future though, but it would require more work on the data package first.

### CODE REVIEW

**Comment by ahocevar** [comment]
I may be missing something, but what happens if the feature's geometry is edited? I'm wondering if it wouldn't be better for now to leave the save operation to the application. There should at least be separate events for validation and server errors, especially given the perspective of doing a refactoring later to leave the save handling to the store. If no protocol is configured, it also does not make much sense to fire a "success" event, because "featuremodified" is also fired. Or am I missing the use case here?

**Comment by bartvde** [comment]
@ahocevar geometry editing was out of scope, but I agree with your other comments.

**Comment by bartvde** [comment]
@ahocevar I've updated the PR per your comments.

File: src/GXM/form/FeatureEditor.js [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+    onFieldChange: function(field, newValue, oldValue) {
+        var feature = this.getFeature(), name = field.getName();
+        if (!feature.modified) {
+            feature.modified = {
+                attributes: {}
+            };
+        }
+        feature.modified.attributes[name] = true;
+        feature.attributes[name] = newValue;
+        feature.state = OpenLayers.State.UPDATE;
```
Comment (ahocevar) [inline src/GXM/form/FeatureEditor.js]: If the feature has a state of `OpenLayers.State.INSERT`, the state should not be set to `OpenLayers.State.UPDATE`.

File: src/GXM/form/FeatureEditor.js [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+    },
+    /**
+     * @private
+     * Listener for the change event on the form fields. Updates the feature.
+     *
+     * @param {Ext.field.Text} this This field
+     * @param {Mixed} newValue The new value
+     * @param {Mixed} oldValue The original value
+     */
+    onFieldChange: function(field, newValue, oldValue) {
```
Comment (ahocevar) [inline src/GXM/form/FeatureEditor.js]: So you modify the feature attributes here, before the user hits the Update button? And when the validation fails, the feature is still modified. Unless I am missing something, I think it makes sense to move this all into `doUpdate()`.

File: src/GXM/form/FeatureEditor.js (hunk 1 above)
Comment (bartvde) [inline src/GXM/form/FeatureEditor.js]: Good point will change.

File: src/GXM/form/FeatureEditor.js (hunk 2 above)
Comment (bartvde) [inline src/GXM/form/FeatureEditor.js]: Agreed, will change.

**Comment by bartvde** [comment]
@ahocevar I've updated the PR per your comments.

**Comment by ahocevar** [comment]
I think this is good to merge now. Thanks for your continued effort on this @bartvde.

**Pull request merged**